src/lib/class/Message/Message.cpp

```diff
@@ -0,0 +1,57 @@
+#include "Message.hpp"
+
+/**
+ * Constructor for the Message class.
+ * Parses a single message from the given string.
+ *
+ * @param msg A constant reference to a std::string containing the message to be parsed.
+ *
+ * @return A new instance of the Message class.
+ */
+Message::Message(std::string const &msg)
+{
+	size_t pos = 0;
```

Comment:
I would've called a function that does the parser instead of doing it all in the constructor